pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp

```diff
@@ -150,11 +150,19 @@ class AdjointJacobian final
             if (ops.hasParams(op_idx)) {
                 if (current_param_idx == *tp_it) {
                     const PrecisionT scalingFactor =
-                        BaseType::applyGenerator(
-                            mu, ops.getOpsName()[op_idx],
-                            ops.getOpsWires()[op_idx],
-                            !ops.getOpsInverses()[op_idx]) *
-                        (ops.getOpsInverses()[op_idx] ? -1 : 1);
+                        (ops.getOpsControlledWires()[op_idx].empty())
+                            ? BaseType::applyGenerator(
+                                  mu, ops.getOpsName()[op_idx],
+                                  ops.getOpsWires()[op_idx],
+                                  !ops.getOpsInverses()[op_idx]) *
+                                  (ops.getOpsInverses()[op_idx] ? -1 : 1)
```

Comment:
Here you are mixing bool operation with int. Try to use only one type

Reply:
```suggestion
                                  ! bool *
                                  (int value diferent from 0 or 1)
```

Reply:
Actually it's multiplying `PrecisionT` with `int`

The `!ops.getOpsInverses()[op_idx]` in L157 is an argument to `applyGenerator`, which returns `PrecisionT`, which is then multiplied by `(ops.getOpsInverses()[op_idx] ? -1 : 1)`

Reply:
ooh sorry, I missed the close bracket before `*`